src/jvm/main/org/jetbrains/kotlinx/lincheck/paramgen/ParameterGenerators.kt

```diff
@@ -0,0 +1,246 @@
+/*
+ * Lincheck
+ *
+ * Copyright (C) 2019 - 2023 JetBrains s.r.o.
+ *
+ * This Source Code Form is subject to the terms of the
+ * Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed
+ * with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.jetbrains.kotlinx.lincheck.paramgen
+
+import org.jetbrains.kotlinx.lincheck.RandomProvider
+import org.jetbrains.kotlinx.lincheck.annotations.Operation
+import java.util.*
+
+/**
+ * The implementation of this interface is used to generate parameters
+ * for [operation][Operation].
+ *
+ * Now all successors must have constructor with [RandomProvider] and config [String].
+ */
+interface ParameterGenerator<T> {
+    fun generate(): T
+
+    /**
+     * Resets current range bounds to start, if this generator has such expanding range.
+     * Meanwhile, it shouldn't reset random to avoid undesired correlation between scenarios.
+     */
+    fun resetRange()
+    class Dummy : ParameterGenerator<Any?> {
```

Comment:
Do you need it? If so, please move it out of the interface and make it an 'object'.